Framework: removed last remaining instance of SectionHeaderButton

rickybanister opened the pull request:
I did several things:

* Switched the 'Make Primary' domain button from the deprecated `SectionHeaderButton` to just a compact `Button`
* Deleted the component finally
* Updated the Devdocs
* Updated the Readme

**Before:**
<img width="747" alt="screen shot 2015-12-10 at 10 50 21 am" src="https://cloud.githubusercontent.com/assets/437258/11720286/9217fb48-9f2c-11e5-9517-676a808752a7.png">

**After:**
<img width="739" alt="screen shot 2015-12-10 at 10 49 50 am" src="https://cloud.githubusercontent.com/assets/437258/11720281/8efca558-9f2c-11e5-87f5-9876bcc42e80.png">


cc @ebinnion @mtias @breezyskies

mtias commented on client/components/section-header/docs/example.jsx:
```diff
@@ -23,16 +23,21 @@ var Cards = React.createClass( {
+						compact>
+						{ this.translate( 'Manage' ) }
+					</Button>
+					<Button
```
Should we make one of these a `primary` one for the example?

mtias commented:
Awesome, thanks for doing this.

breezyskies commented:
`Make Primary` button looks good. :+1:

rickybanister commented:
I fixed some css and updated the devdocs to show a primary action:

<img width="994" alt="screen shot 2015-12-10 at 11 18 49 am" src="https://cloud.githubusercontent.com/assets/437258/11721044/dade23ae-9f2f-11e5-86c6-a22bcf021b76.png">

rickybanister commented:
cc @shaunandrews as this affects #1391

ebinnion commented on client/components/section-header/docs/example.jsx:
```diff
@@ -23,16 +23,26 @@ var Cards = React.createClass( {
+					<Button
+						compact
+						primary>
+						{ this.translate( 'Primary Action' ) }
```
I could be wrong, but I don't think we need to translate the strings in examples. If we do keep the translation, we should probably at least provide some context. Maybe something like,  `this.translate( 'Primary Action', { context: 'Button display in documentation' } )`?

cc @deBhal who might know whether we need to translate these strings.

__Note:__ There's also another translated string in this file on line `33`.

ebinnion commented:
I'm not sure what route to use to test this component: `client/my-sites/upgrades/domain-management/edit/card/header/primary-domain-button.jsx`. I'm glad to test if you can help me out with that.

Code-wise, this LGTM, except comment about not using translated strings in example components, but I'm not sure which way that should go.

rickybanister commented:
Please update the examples to fit the standard, I just saw translations in one file and not the other so I made it consistent. The contexts probably make sense too.

To test the domain button you have to visit a site that has a custom domain, then click on the domains sidebar item, then click on the non primary domain (i.e. example.wordpress.com) 

The url would be something like:

`https://wpcalypso.wordpress.com/domains/manage/example.wordpress.com/edit/example.com`

ebinnion commented:
I went ahead and removed the translated strings. I also was able to test that the domain management section header worked fine. :+1:

deBhal commented:
Sorry for the late reply, I've been out of commission.

These strings won't get picked up or translated, so there's no concern over wasting translator's time, it's just a matter of communications with developers. (We scan for strings after we bundle up the code, so we skip docs and tests automagically.  You can always `make translate` and check `calypso-strings.js` in the top level directory if you're not sure).

So it's just a question of whether it's better example code or not.  I'd lean towards including the translate calls in general, unless you feel like they obscure the point of the example.

I feel like in this example, you gain a little, particularly with the label attribute where you get a bit of extra signalling that this is a user-facing string and not a control value.

I've opened PR #1536 because I feel bad about being away and missing the other PR, not to be pushy.  If you feel like the example stands better without the changes, feel free to comment to that effect and close it.
